components/board/genericlinux/gpio.go

```diff
@@ -7,28 +7,46 @@ package genericlinux
 import (
 	"context"
 	"fmt"
-	"math"
 	"sync"
+	"time"
 
+	"github.com/edaniels/golog"
 	"github.com/mkch/gpio"
 	"github.com/pkg/errors"
+	goutils "go.viam.com/utils"
 
 	"go.viam.com/rdk/components/board"
 )
 
 type gpioPin struct {
-	// These values should both be considered immutable. The mutex is only here so that the use of
-	// the multiple calls to the gpio package don't have race conditions.
+	// These values should both be considered immutable.
 	devicePath string
 	offset     uint32
-	mu         sync.Mutex
+
+	// These values are mutable. Lock the mutex when interacting with them.
+	pwmRunning      bool
+	pwmFreqHz       uint
+	pwmDutyCyclePct float64
+
+	mu        sync.Mutex
+	cancelCtx context.Context
+	waitGroup *sync.WaitGroup
+	logger    golog.Logger
 }
 
 // This helps implement the board.GPIOPin interface for gpioPin.
 func (pin *gpioPin) Set(ctx context.Context, isHigh bool, extra map[string]interface{}) (err error) {
 	pin.mu.Lock()
 	defer pin.mu.Unlock()
 
+	pin.pwmRunning = false
+	return pin.setInternal(isHigh)
+}
+
+// This function assumes you've already locked the mutex. It does not modify pwmRunning because we
```

Comment:
I think the fact that there's a separate SetPWM function in our pin interface is enough to explain why this function doesn't modify `pwmRunning`. Although it's still worth noting that this is used by the software PWM logic as well